common/scala/src/main/scala/org/apache/openwhisk/core/database/CouchDbRestStore.scala

```diff
@@ -166,10 +166,20 @@ class CouchDbRestStore[DocumentAbstraction <: DocumentSerializer](dbProtocol: St
       }
     }
 
-    reportFailure(
-      f,
-      failure =>
-        transid.failed(this, start, s"[PUT] '$dbName' internal error, failure: '${failure.getMessage}'", ErrorLevel))
+    f.onComplete {
+      case Failure(_: ArtifactStoreException) => // These failures are intentional and shouldn't trigger the catcher.
+      case Failure(_) if retry > 0 =>
+        transid.failed(this, start, s"store activation to CouchDB failed, retry after ${retryDelay}")
+        system.scheduler.scheduleOnce(retryDelay) {
```

Comment:
* it is better to use transid.failed here
* It is better to add some delay when retry happenes.